zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+        self.upload_tasks.add_task(task)
+        ImageUploadDaemon(task).start()
+
+    def _get_upload_path(self, req):
+        host = req[http.REQUEST_HEADER]['Host']
+        return 'http://' + host + self.UPLOAD_IMAGE_PATH
+
+    def get_upload_progress(self, req):
+        cmd = jsonobject.loads(req[http.REQUEST_BODY])
+        task = self.upload_tasks.get_task(cmd.imageUuid)
+        if task is None:
+            raise Exception('image not found %s' % cmd.imageUuid)
```

Ruff (0.13.1)

Create your own exception

(TRY002)